src/elementpenalty.jl

```diff
@@ -118,12 +115,12 @@ immutable ElasticNetPenalty{T <: Number} <: ConvexElementPenalty
         new{T}(α)
     end
 end
-for f in [:value, :deriv]
-    @eval function ($f){T <: Number}(p::ElasticNetPenalty{T}, θ::T)
-        p.α * ($f)(L1Penalty(), θ) + (1 - p.α) * ($f)(L2Penalty(), θ)
+for f in (:value, :deriv)
+    @eval function ($f){T <: Number}(p::ElasticNetPenalty{T}, θ::Number)
+        p.α * ($f)(L1Penalty(), θ) + (one(T) - p.α) * ($f)(L2Penalty(), θ)
```

Comment:
probably not for the default `Number` types in base. But `Number` can be extended by user packages (for example `DualNumbers`) for which we don't know if `Int` is a neutral type. This seems like the cleanest option to be as user friendly as possible. That said, probably 99% of people use these functions with `Float64`, so I doubt anyone runs into an issue in practice